nilearn/plotting/surf_plotting.py

```diff
@@ -29,10 +31,502 @@
+        )
+    return surf_map_faces
+
+
+def _configure_cmap_matplotlib(cmap, vmin, vmax, threshold,
+                               cbar_tick_format):
+    """Helper function for plot_surf with matplotlib engine.
+
+    This function configure the colormap.
+    """
+    our_cmap = get_cmap(cmap)
+    norm = Normalize(vmin=vmin, vmax=vmax)
+    # Default number of ticks is 5...
+    nb_ticks = 5
+    # ...unless we are dealing with integers with a small range
+    # in this case, we reduce the number of ticks
+    if cbar_tick_format == "%i" and vmax - vmin < nb_ticks:
+        ticks = np.arange(vmin, vmax + 1)
+        nb_ticks = len(ticks)
+    else:
+        ticks = np.linspace(vmin, vmax, nb_ticks)
+    bounds = np.linspace(vmin, vmax, our_cmap.N)
+    if threshold is not None:
+        cmaplist = [our_cmap(i) for i in range(our_cmap.N)]
+        # set colors to grey for absolute values < threshold
+        istart = int(norm(-threshold, clip=True) * (our_cmap.N - 1))
+        istop = int(norm(threshold, clip=True) * (our_cmap.N - 1))
+        for i in range(istart, istop):
+            cmaplist[i] = (0.5, 0.5, 0.5, 1.)
+        our_cmap = LinearSegmentedColormap.from_list(
+            'Custom cmap', cmaplist, our_cmap.N)
+    return our_cmap, norm, ticks, bounds
+
+
+def _compute_facecolors(bg_map, faces, n_vertices, darkness, alpha):
+    """Helper function for plot_surf.
+
+    This function computes the facecolors.
+    """
+    # set alpha if in auto mode
+    if alpha == 'auto':
+        alpha = .5 if bg_map is None else 1
+    face_colors = np.ones((faces.shape[0], 4))
+    if bg_map is None:
+        bg_data = np.ones(n_vertices) * 0.5
+    else:
+        bg_data = load_surf_data(bg_map)
+        if bg_data.shape[0] != n_vertices:
+            raise ValueError('The bg_map does not have the same number '
+                             'of vertices as the mesh.')
+    bg_faces = np.mean(bg_data[faces], axis=1)
+    if bg_faces.min() != bg_faces.max():
+        bg_faces = bg_faces - bg_faces.min()
+        bg_faces = bg_faces / bg_faces.max()
+    # control background darkness
+    bg_faces *= darkness
+    face_colors = plt.cm.gray_r(bg_faces)
+    # modify alpha values of background
+    face_colors[:, 3] = alpha * face_colors[:, 3]
+    return face_colors
+
+
+def _threshold_and_rescale(data, threshold, vmin, vmax):
+    """Helper function for plot_surf.
+
+    This function thresholds and rescales the provided data.
+    """
+    data_copy = np.copy(data)
+    # if no vmin/vmax are passed figure them out from data
+    if vmin is None:
+        vmin = np.nanmin(data_copy)
+    if vmax is None:
+        vmax = np.nanmax(data_copy)
+    # treshold if indicated
+    if threshold is None:
+        # If no thresholding and nans, filter them out
+        kept_indices = np.where(
+            np.logical_not(
+                np.isnan(data_copy)))[0]
+    else:
+        kept_indices = np.where(np.abs(data_copy) >= threshold)[0]
```

Comment:
why is interesting to have `kept_indices` as indices rather than a binary vector ?

Reply:
@bthirion I agree. In the majority of cases it should be more compact and efficient to use a binary vector here (and the code is easier to read as a bonus...).